### PR TITLE
one Gemfile.lock for all adapters

### PR DESCRIPTION
**Pull request by jouve**

### CODE REVIEW

File: Dockerfile-debian.template [hunk 1]
```diff
@@ -114,13 +114,12 @@ RUN set -eux; \
 	; \
 	rm -rf /var/lib/apt/lists/*; \
 	\
-	gosu redmine bundle install --jobs "$(nproc)" --without development test; \
+	echo > ./config/database.yml; \
```
Comment (tianon) [inline Dockerfile-debian.template:117]:
```suggestion
# fill up "database.yml" with bogus entries so the redmine Gemfile will pre-install all database adapter dependencies
# https://github.com/redmine/redmine/blob/e9f9767089a4e3efbd73c35fc55c5c7eb85dd7d3/Gemfile#L50-L79
	echo '# the following entries only exist to force `bundle install` to pre-install all database adapter dependencies -- they can be safely removed/ignored' > ./config/database.yml; \
```

**Review by tianon — COMMENTED** [review]
Very interesting workaround!  I've left one suggestion that I think will help improve the discoverability of _why_ this works (and provide a hint to users who might open the image and inspect `config/database.yml` and then wonder why it's so strange).

File: Dockerfile-debian.template (hunk 1 above)
Comment (jouve) [reply Dockerfile-debian.template:117]: the database.yml is removed in the same layer :)

**Review by yosifkit — APPROVED** [review]
> +30 −84

:+1:

**Review by tianon — APPROVED** [review]

**Pull request merged**